docs/src/_reference/command-line-interface.md

```diff
@@ -604,6 +604,12 @@ meltano invoke --print-var <PLUGIN_ENVIRONMENT_VARIABLE_1> <PLUGIN_NAME>
 meltano invoke --print-var <PLUGIN_ENVIRONMENT_VARIABLE_1> --print-var <PLUGIN_ENVIRONMENT_VARIABLE_2> <PLUGIN_NAME>
 ```
 
+## `lock`
+
+`meltano lock` creates lock files for all [non-custom](/concepts/plugins#custom-plugins) plugins in the project.
```

Comment:
@edgarrmondragon per https://github.com/meltano/meltano/discussions/6018 and https://github.com/meltano/meltano/issues/5961#issuecomment-1146682249 we should have `--all` be the default flag so that `meltano lock` and `meltano lock --all` are equivalent.

Reply:
@tayloramurphy I've updated the docs to make usage options more clear: https://deploy-preview-6114--meltano.netlify.app/reference/command-line-interface#lock

So, to confirm: we want an `--all` flag that has no effect at the moment, i.e. `meltano lock --all` is exactly the same as `meltano lock`.

Reply:
@edgarrmondragon - Thanks for confirming - yes, that's the spec we were leaning towards, from this discussion: https://github.com/meltano/meltano/discussions/6018

I don't think this is a closed/settled issue, per se, but I felt good about what we were outlining in that discussion

I think there are two categories of commands:

1. Those that should default to noop (actually exit code 1) if no direct targets are passed. (e.g. `meltano run` as a silly example, but `meltano config set` as a more realistic one).
2. Those that should default to '--all' if no direct targets are passed.

What we think the reasonable default is of the above, may not be obvious to the user, and it could also change over time. For instance, we don't have `meltano config get --all` but in future, we could! To avoid writing a convention that changes or breaks according to whether `--all` is viable in the context, we were biasing towards accepting `--all` as an explicit way to say "yes, I really mean everything".

With the above context and framing, `meltano lock` could fail on it's own or succeed according to the decisions we make here in this PR, but `meltano lock --all` would be an intuitive way to run everything without having to remember the outcome of our decision on this point 😄 .

So, I'm actually fine with either of these behaviors:

- `meltano lock` is a noop with exit code 1
- `meltano lock` is a synonym for `meltano lock --all`

But `meltano lock my-plugin` definitely does not do `--all` and meltano lock --all my-plugin` is an illegal operation because `--all` and the named plugin are mutually exclusive.

Reply:
@aaronsteers

> But `meltano lock my-plugin` definitely does not do the "--all" behavior,

Ok, so the current interface in this PR uses `meltano lock <type>` and `meltano lock <type> <name> <name_two>`. Do we want to remove the `<type>` argument? I added it to make the command consistent with `meltano install` and friends.

> `meltano lock` is a noop with exit code 1

I like that 👍

Reply:
@edgarrmondragon - Sorry, yes. I think if it is not too late, I think _not_ having to declare the plugin type unless there's a naming conflict is probably the best path forward for new CLI commands.

Reply:
@aaronsteers sounds good, definitely not too late 🙂. I had forgotten that @cjohnhanson added a reference to clig.dev in the handbook:

> **Prefer flags to args.** It’s a bit more typing, but it makes it much clearer what is going on. It also makes it easier to make changes to how you accept input in the future. Sometimes when using args, it’s impossible to add new input without breaking existing behavior or creating ambiguity.

So this seems to be a good decision.

Reply:
@edgarrmondragon - Sorry we had to discover this after dev started - I dropped this discussion to try to more proactively and formally handle this in future developments: https://github.com/meltano/meltano/discussions/6222